### PR TITLE
Unified tests

### PR DESCRIPTION
**Pull request by belka-ew**
This is the first step to restructure dmocks.

The unittests are currently hidden in a set of version blocks like:
- DMocksTest
- DMocksDebug
- DMocksTestStandalone

So if you run `dub test` without `--config DMocksTest`, the tests are just skipped.
This PR removes these unittest version blocks, so the tests aren't skipped. It also comments out several tests that were already broken in the original DMocks-revived repository and probably should be reviewed later.

### CODE REVIEW

**Review by linkrope — CHANGES_REQUESTED** [review]
- can you remove the obsolete `test` template?
- can you name `unittest` functions `@("like this")`
  (works with unit-threaded, but does not require unit-threaded)
- why `private unittest`?

**Comment by belka-ew** [comment]
1). `test` was removed before its usage: [fbab160](https://github.com/funkwerk/DMocks-revived/commit/fbab16036f4e42be69f1d13f39d6c24a61922ed5)
3). [Documented Unittests](https://dlang.org/spec/unittest.html#documented-unittests):
> A unittest which is not documented, or is marked as private will not be used to generate code samples.

And I skipped a large DMocksTest-block. I'll fix it with the next commit.

File: dmocks/mocks.d [hunk 1]
```diff
@@ -1191,76 +629,669 @@ version (DMocksTest)
...
+@("expect")
+unittest
+{
+    Mocker m = new Mocker();
+    Object o = m.mock!(Object);
+    m.expect(o.toString).repeat(0).returns("mrow?");
+    m.replay();
+    try {
+        o.toString;
+    } catch (Exception e) {}
```
Comment (linkrope) [inline dmocks/mocks.d]: `assert(false, ...);` is missing?
Consider using `assertThrown` instead.

**Review by linkrope — CHANGES_REQUESTED** [review]
ugly indentation for optional `msg` of `assertThrown`
(default message "assertThrown failed: No ... was thrown." is good enough)

**Comment by belka-ew** [comment]
@linkrope, do you mean just remove "Expected exception not thrown"?

**Review by linkrope — APPROVED** [review]

**Pull request merged**